packages/zoltan2/CMakeLists.txt

```diff
@@ -68,23 +68,34 @@ TRIBITS_ADD_OPTION_AND_DEFINE(
   OFF )
 
 TRIBITS_ADD_OPTION_AND_DEFINE(
-   ${PACKAGE_NAME}_ENABLE_UseZoltanComm
-   ${PACKAGE_NAME_UC}_USEZOLTANCOMM
-   "For migration in MJ, use Zoltan's comm package instead of Tpetra."
-   OFF)
+  ${PACKAGE_NAME}_ENABLE_UseZoltanComm
+  ${PACKAGE_NAME_UC}_USEZOLTANCOMM
+  "For migration in MJ, use Zoltan's comm package instead of Tpetra."
+  OFF)
 
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  ${PACKAGE_NAME}_ENABLE_Machine_Dragonfly
```

Comment:
Why does this need to be a compile-time option?  It seems like the choice could be a run-time parameter.  I'm not a fan of the proliferation of compile-time options; that, as well as hours-long compile times, make testing more difficult than it already is.

Reply:
We will address this problem in #5927.  It is relevant to our unit-testing of various "dummy" machines within a single build.  We expect to resolve #5927 within a few weeks, but would like to push this PR in the meantime to allow @MicheldeMessieres to begin merging with #5303 .
 Thanks, @mhoemmen

Reply:
Thanks @kddevin !

Reply:
Resolved since Zoltan2 plans to address this issue later.